kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -435,11 +433,17 @@ public Map<String, String> addTags(String account, String container, Map<String,
 
 
 		} catch (Exception e) {
-            connection = null;
-			LOGGER.error(SESSIONID, REGISTRATIONID, "Exception occured while addTags for : " + container,
-					ExceptionUtils.getStackTrace(e));
-			throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(),
-					OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
+            // this check is introduced to support backward compatibility
+            if (e instanceof AmazonS3Exception && e.getMessage().contains(TAG_BACKWARD_COMPATIBILITY_ERROR)) {
+                connection.deleteObject(bucketName, finalObjectName);
+                addTags(account, container, tags);
+            } else {
+                connection = null;
+                LOGGER.error(SESSIONID, REGISTRATIONID, "Exception occured while addTags for : " + container,
+                        ExceptionUtils.getStackTrace(e));
+                throw new ObjectStoreAdapterException(OBJECT_STORE_NOT_ACCESSIBLE.getErrorCode(),
+                        OBJECT_STORE_NOT_ACCESSIBLE.getErrorMessage(), e);
+            }
```

Comment:
I think we need to handle this in the get tags methods also, since that will called at the end of packet uploader stage even before packet goes to packet classifier stage to add tags.